README
Project information:

open-vm-tools <http://open-vm-tools.sourceforge.net/>

	These are the release notes for the open-vm-tools.  Read them carefully, as
they explain how to build this project for different platforms and various 
different Linux distributions.

================================================================================
General information:

(*)What are the open-vm-tools? 
	The open-vm-tools are a subset of the VMware Tools, currently composed of
kernel modules for Linux and user-space programs for all VMware supported Unix
like guest operating systems.

(*)Where do I find documentation for the open-vm-tools beyond this README?
	Please refer to the main project web site for all the latest documentation 
at: <http://open-vm-tools.sourceforge.net/> 

(*)How do I build the open-vm-tools?
	The open-vm-tools uses the GNU Automake tool for generating Makefiles to 
build all sources.  More information about Automake can be found here: 
<http://sources.redhat.com/automake/>

For more information on building this project see the "Project build
information" section of this document. 

================================================================================
Project build information:

(*)Getting configure options & help:
	If you are looking for help or additional settings for the building of this
project the following configure command will display a list of help options:
	"./configure --help"

(*)Using configure:
	When using configure in the steps below it is only necessary to call
"./configure" once unless there was a problem after the first invocation.

(*)Building Unix user-space programs:
1) "./configure"
2) Run "make" to build Unix userland programs
3) Run "make modules" to build kernel modules

(*)Why is there no "make install" make target?
	In their current state, the various Tools applications and modules require a
fair bit of installation support to run properly.  VMware currently maintains a
Perl-based installer for the Tools that is pretty good about providing that
support under a variety of distributions and environments.  It doesn't make
sense to rewrite this installer under make; distribution-specific Tools
packages (tailored for a particular target environment) are better suited to
the task.  If you are interested in creating a Tools package, please see
<http://open-vm-tools.sourceforge.net/> for more information.

================================================================================
Build suggestions:
	Each Linux distribution puts binaries, libraries and development headers in 
different locations.  This leads to having to setup different paths for the
compiler and linker for the Automake system.  Below is a list of known
configurations to build for platforms that do not "just work" out of the box
or the tar-ball as the case may be.

General Build note:
	Please make note that the "--disable-multimon" flag should only be used when
there are no libXinerama libraries installed, or you do not want to enable multi
monitor support.

	Your operating system version will influence which kernel
modules can be built. Here are the minimum OS versions for each
module:

|------------------------------------------------------------------------------|
| vmsync        | vmblock        | vmxnet       | vmmemctl       | vmhgfs      |
|---------------|----------------|--------------|----------------|-------------|
| Linux 2.6.6   | Linux 2.4.0    | Linux 2.2.0  | Linux 2.2.0    | Linux 2.4.0 |
| FreeBSD N/A   | FreeBSD 6.0    | FreeBSD 4.9  | FreeBSD 3.2    | FreeBSD 6.0 |
|------------------------------------------------------------------------------|

(*)Suse Enterprise 10.1 (i386 & x86_64):
i386:
./configure LDFLAGS="-L/opt/gnome/lib"

x86_64:
./configure LDFLAGS="-L/opt/gnome/lib64"

(*)OpenSolaris "Nevada" (i386 & x86_64):
	Before building open-vm-tools on Solaris, you will need to make 
sure that libdnet (http://libdnet.sourceforge.net/) is installed and 
that the 'dnet-config' script is on your shell's PATH.

	Make sure that the GNU tools are in your shell's PATH. The 
default GNU tools location on OpenSolaris is: "/usr/sfw/bin". Once the 
PATH is set in your build shell then the following configure command 
should work:

./configure --disable-multimon

(*)FreeBSD 6.2 (i386 & x86_64):
./configure LDFLAGS="-L/usr/local/lib"

================================================================================
Other resources:
	There are also open-source video & mouse drivers for Xorg/XFree86 that
provide better performance and additional features to Linux, Solaris, and BSD
derivatives guests running X11. 

	The drivers 'xf86-input-vmmouse' and 'xf86-video-vmware' are included with 
recent Xorg releases, and it shouldn't be necessary to do anything extra to 
obtain them.  If you want to obtain them separately, they can be downloaded at:
<http://xorg.freedesktop.org/releases/individual/driver/>

================================================================================
Guest operating systems used to test on this release:

(*)Fedora Core 8 (i386 & x86_64)
(*)Redhat 9.0 (i386)
(*)Redhat Enterprise 4 U5 (i386 & x86_64)
(*)Redhat Enterprise 5 (i386 & x86_64)
(*)FreeBSD 5.4 (i386 & x86_64)
(*)FreeBSD 6.2 (i386 & x86_64)
(*)OpenSolaris 10 "Nevada" (i386 & x86_64)
(*)Open SuSE 10.3 (i386 & x86_64)
(*)Ubuntu 6.04 (i386 & x86_64)
(*)Ubuntu 7.10 (i386 & x86_64)

===============================================================================
Known issues:

(*) When using HGFS on FreeBSD, the 'cp' command will fail if the source 
file is on the HGFS share. This is because mmap() does not yet work with 
the FreeBSD port of HGFS. One workaround is to use 'cat 
/mnt/hgfs/foo/bar > /tmp/baz' instead of 'cp /mnt/hgfs/foo/bar 
/tmp/baz'.

(*) When using HGFS on FreeBSD, if your current working directory is the 
top-level HGFS mount (e.g. '/mnt/hgfs') and you run 'ls', you will 
receive an "Invalid argument" error. The workaround is to change to 
another directory and then run 'ls /mnt/hgfs'. Running 'ls' in 
subdirectories of /mnt/hgfs works as expected.

===============================================================================


debian/changelog
open-vm-tools (1:9.0.0+vyos1+helium2) kernel; urgency=low

  * Eliminate remaining Vyatta names.

 -- Daniil Baturin <daniil@baturin.org>  Tue, 05 Aug 2014 16:17:49 +0000

open-vm-tools (1:9.0.0+vyos1+helium1) kernel; urgency=low

  * New branch

 -- Daniil Baturin <daniil@baturin.org>  Sat, 15 Feb 2014 15:32:34 +0100

open-vm-tools (1:9.0.0+vyatta+hydrogen1) kernel; urgency=low

  * New branch

 -- Daniil Baturin <daniil@baturin.org>  Sat, 16 Nov 2013 23:54:54 +0100

open-vm-tools (1:9.0.0+vyatta+daisy3) kernel; urgency=low

  * Add missing files

 -- James Davidson <james.davidson@vyatta.com>  Tue, 12 Mar 2013 16:09:20 -0700

open-vm-tools (1:9.0.0+vyatta+daisy2) kernel; urgency=low

  [ James Davidson ]
  * Update to open-vm-tools 9.0.0

  [ Stephen Hemminger ]
  * Fix linux include path
  * fix modules install path

  [ James Davidson ]

 -- James Davidson <james.davidson@vyatta.com>  Tue, 12 Mar 2013 14:53:50 -0700

open-vm-tools (1:9.0.0+vyatta+daisy1) kernel; urgency=low

  * Upgrade to open-vm-tools 9.0.0

 -- James Davidson <james.davidson@vyatta.com>  Wed, 06 Mar 2013 12:56:02 -0800

open-vm-tools (1:8.8.2+vyatta-10+daisy4) kernel; urgency=low

  * Force rebuild; kernel changed

 -- James Davidson <james.davidson@vyatta.com>  Thu, 20 Dec 2012 12:18:43 -0800

open-vm-tools (1:8.8.2+vyatta-10+daisy3) kernel; urgency=low

  * force release: kernel patch to allow conntrack leak detection

 -- Gaurav Sinha <gaurav.sinha@vyatta.com>  Fri, 14 Dec 2012 13:38:19 -0800

open-vm-tools (1:8.8.2+vyatta-10+daisy2) kernel; urgency=low

  * Force rebuild

 -- James Davidson <james.davidson@vyatta.com>  Fri, 26 Oct 2012 20:51:38 -0700

open-vm-tools (1:8.8.2+vyatta-10+daisy1) kernel; urgency=low

  * create daisy branch

 -- John Southworth <john.southworth@vyatta.com>  Sat, 13 Oct 2012 13:30:01 -0700

open-vm-tools (1:8.8.2+vyatta-10) kernel; urgency=low

  * new branch

 -- John Southworth <john.southworth@vyatta.com>  Fri, 12 Oct 2012 19:45:48 -0700

open-vm-tools (1:8.8.2+vyatta-9) kernel; urgency=low

  * Rebuild for kernel patch

 -- John Southworth <john.southworth@vyatta.com>  Fri, 05 Oct 2012 12:04:47 -0700

open-vm-tools (1:8.8.2+vyatta-8) unstable; urgency=low

  * Force rebuild for new kernel

 -- John Southworth <john.southworth@vyatta.com>  Thu, 13 Sep 2012 17:08:44 -0700

open-vm-tools (1:8.8.2+vyatta-7) unstable; urgency=low

  * Merge 64bit standard and virt iso builds

 -- John Southworth <john.southworth@vyatta.com>  Wed, 15 Aug 2012 10:20:51 -0700

open-vm-tools (1:8.8.2+vyatta-6) unstable; urgency=low

  * Force rebuild; kernel changed

 -- James Davidson <james.davidson@vyatta.com>  Wed, 01 Aug 2012 18:55:52 -0700

open-vm-tools (1:8.8.2+vyatta-5) unstable; urgency=low

  * Increase max NICs to 256 to avoid assert

 -- James Davidson <james.davidson@vyatta.com>  Thu, 21 Jun 2012 19:04:42 -0700

open-vm-tools (1:8.8.2+vyatta-4) unstable; urgency=low

  * force rebuild

 -- Stephen Hemminger <shemminger@vyatta.com>  Thu, 21 Jun 2012 11:08:45 -0700

open-vm-tools (1:8.8.2+vyatta-3) unstable; urgency=low

  * force rebuild

 -- Stephen Hemminger <shemminger@vyatta.com>  Thu, 14 Jun 2012 16:43:21 -0700

open-vm-tools (1:8.8.2+vyatta-2) unstable; urgency=low

  * update to match changed kernel config

 -- Stephen Hemminger <shemminger@vyatta.com>  Tue, 12 Jun 2012 14:47:20 -0700

open-vm-tools (1:8.8.2+vyatta-1) unstable; urgency=low

  * Update to open-vm-tools 8.8.2
  * Compile without procps on amd64
  * vmci: fix warnings
  * Add umode_t typedef

 -- Stephen Hemminger <shemminger@vyatta.com>  Mon, 04 Jun 2012 17:53:36 -0700

open-vm-tools (1:8.8.0+vyatta-33) unstable; urgency=low

  * Update to 3.3.8 kernel

 -- Stephen Hemminger <stephen.hemminger@vyatta.com>  Mon, 04 Jun 2012 10:44:09 -0700

open-vm-tools (1:8.8.0+vyatta-32) unstable; urgency=low

  * Fix snapshot when same filesystem mounted multiple times

 -- Stephen Hemminger <shemminger@vyatta.com>  Fri, 01 Jun 2012 12:47:45 -0700

open-vm-tools (1:8.8.0+vyatta-31) unstable; urgency=low

  * update to 3.3.7 kernel

 -- Stephen Hemminger <shemminger@vyatta.com>  Tue, 22 May 2012 09:52:35 -0700

open-vm-tools (1:8.8.0+vyatta-30) unstable; urgency=low

  * force rebuild

 -- Stephen Hemminger <shemminger@vyatta.com>  Tue, 15 May 2012 14:40:19 -0700

open-vm-tools (1:8.8.0+vyatta-29) unstable; urgency=low

  * update to 3.3.6 kernel

 -- Stephen Hemminger <shemminger@vyatta.com>  Tue, 15 May 2012 09:02:26 -0700

open-vm-tools (1:8.8.0+vyatta-28) unstable; urgency=low

  * update to 3.3.5 kernel

 -- Stephen Hemminger <shemminger@vyatta.com>  Tue, 08 May 2012 15:58:50 -0700

open-vm-tools (1:8.8.0+vyatta-27) unstable; urgency=low

  * update to 3.3.4 kernel

 -- Stephen Hemminger <shemminger@vyatta.com>  Fri, 27 Apr 2012 12:20:29 -0700

open-vm-tools (1:8.8.0+vyatta-26) unstable; urgency=low

  * update to 3.3.3 kernel

 -- Stephen Hemminger <shemminger@vyatta.com>  Tue, 24 Apr 2012 14:13:40 -0700

open-vm-tools (1:8.8.0+vyatta-25) unstable; urgency=low

  * update to 3.3.2

 -- Stephen Hemminger <shemminger@vyatta.com>  Fri, 13 Apr 2012 16:49:52 -0700

open-vm-tools (1:8.8.0+vyatta-24) unstable; urgency=low

  * Fix build

 -- Stephen Hemminger <stephen.hemminger@vyatta.com>  Tue, 03 Apr 2012 07:51:15 -0700

open-vm-tools (1:8.8.0+vyatta-23) unstable; urgency=low

  * Update to kernel 3.3.1

 -- Stephen Hemminger <stephen.hemminger@vyatta.com>  Mon, 02 Apr 2012 21:04:16 -0700

open-vm-tools (1:8.8.0+vyatta-22) unstable; urgency=low

  * 1:8.8.0+vyatta-22

 -- Arthur Xiong <arthur@vyatta.com>  Mon, 02 Apr 2012 10:52:42 -0700

open-vm-tools (1:8.8.0+vyatta-21) unstable; urgency=low

  * Allow building open-vm-tools against any flavor kernel
  * vmxnet: update to latest network driver api
  * Update to 3.3.0 kernel version dependency
  * vmxnet: update to skb frag API
  * hgfs: override warning about i_nlink
  * hgfs: add module.h
  * hgfs: remove fsync


libguestlib/guestlibV3.h
/*
 * Please do not edit this file.
 * It was generated using rpcgen.
 */

#ifndef _GUESTLIBV3_H_RPCGEN
#define _GUESTLIBV3_H_RPCGEN

#include <vmxrpc.h>


#ifdef __cplusplus
extern "C" {
#endif


struct GuestLibV3StatUint32 {
	Bool valid;
	uint32 value;
};
typedef struct GuestLibV3StatUint32 GuestLibV3StatUint32;

struct GuestLibV3StatUint64 {
	Bool valid;
	uint64 value;
};
typedef struct GuestLibV3StatUint64 GuestLibV3StatUint64;
#define GUESTLIBV3_STRING_SIZE 512

struct GuestLibV3String {
	Bool valid;
	char *value;
};
typedef struct GuestLibV3String GuestLibV3String;

struct GuestLibV3ByteArray {
	Bool valid;
	struct {
		u_int value_len;
		char *value_val;
	} value;
};
typedef struct GuestLibV3ByteArray GuestLibV3ByteArray;

typedef uint32 GuestLibV3StatCount;

enum GuestLibV3TypeIds {
	GUESTLIB_TYPE_RESERVED = 0,
	GUESTLIB_CPU_RESERVATION_MHZ = 1,
	GUESTLIB_CPU_LIMIT_MHZ = 2,
	GUESTLIB_CPU_SHARES = 3,
	GUESTLIB_CPU_USED_MS = 4,
	GUESTLIB_HOST_MHZ = 5,
	GUESTLIB_MEM_RESERVATION_MB = 6,
	GUESTLIB_MEM_LIMIT_MB = 7,
	GUESTLIB_MEM_SHARES = 8,
	GUESTLIB_MEM_MAPPED_MB = 9,
	GUESTLIB_MEM_ACTIVE_MB = 10,
	GUESTLIB_MEM_OVERHEAD_MB = 11,
	GUESTLIB_MEM_BALLOONED_MB = 12,
	GUESTLIB_MEM_SWAPPED_MB = 13,
	GUESTLIB_MEM_SHARED_MB = 14,
	GUESTLIB_MEM_SHARED_SAVED_MB = 15,
	GUESTLIB_MEM_USED_MB = 16,
	GUESTLIB_ELAPSED_MS = 17,
	GUESTLIB_RESOURCE_POOL_PATH = 18,
	GUESTLIB_CPU_STOLEN_MS = 19,
	GUESTLIB_MEM_TARGET_SIZE_MB = 20,
	GUESTLIB_HOST_CPU_NUM_CORES = 21,
	GUESTLIB_HOST_CPU_USED_MS = 22,
	GUESTLIB_HOST_MEM_SWAPPED_MB = 23,
	GUESTLIB_HOST_MEM_SHARED_MB = 24,
	GUESTLIB_HOST_MEM_USED_MB = 25,
	GUESTLIB_HOST_MEM_PHYS_MB = 26,
	GUESTLIB_HOST_MEM_PHYS_FREE_MB = 27,
	GUESTLIB_HOST_MEM_KERN_OVHD_MB = 28,
	GUESTLIB_HOST_MEM_MAPPED_MB = 29,
	GUESTLIB_HOST_MEM_UNMAPPED_MB = 30,
	GUESTLIB_MEM_ZIPPED_MB = 31,
	GUESTLIB_MEM_ZIPSAVED_MB = 32,
	GUESTLIB_MEM_LLSWAPPED_MB = 33,
	GUESTLIB_MEM_SWAP_TARGET_MB = 34,
	GUESTLIB_MEM_BALLOON_TARGET_MB = 35,
	GUESTLIB_MEM_BALLOON_MAX_MB = 36,
	GUESTLIB_RESOURCE_POOL_PATH_LONG = 37,
	GUESTLIB_MAX_STATISTIC_ID = 38,
};
typedef enum GuestLibV3TypeIds GuestLibV3TypeIds;

struct GuestLibV3Stat {
	GuestLibV3TypeIds d;
	union {
		struct GuestLibV3StatUint32 cpuReservationMHz;
		struct GuestLibV3StatUint32 cpuLimitMHz;
		struct GuestLibV3StatUint32 cpuShares;
		struct GuestLibV3StatUint64 cpuUsedMs;
		struct GuestLibV3StatUint32 hostMHz;
		struct GuestLibV3StatUint32 memReservationMB;
		struct GuestLibV3StatUint32 memLimitMB;
		struct GuestLibV3StatUint32 memShares;
		struct GuestLibV3StatUint32 memMappedMB;
		struct GuestLibV3StatUint32 memActiveMB;
		struct GuestLibV3StatUint32 memOverheadMB;
		struct GuestLibV3StatUint32 memBalloonedMB;
		struct GuestLibV3StatUint32 memSwappedMB;
		struct GuestLibV3StatUint32 memSharedMB;
		struct GuestLibV3StatUint32 memSharedSavedMB;
		struct GuestLibV3StatUint32 memUsedMB;
		struct GuestLibV3StatUint64 elapsedMs;
		struct GuestLibV3String resourcePoolPath;
		struct GuestLibV3StatUint64 cpuStolenMs;
		struct GuestLibV3StatUint64 memTargetSizeMB;
		struct GuestLibV3StatUint32 hostCpuNumCores;
		struct GuestLibV3StatUint64 hostCpuUsedMs;
		struct GuestLibV3StatUint64 hostMemSwappedMB;
		struct GuestLibV3StatUint64 hostMemSharedMB;
		struct GuestLibV3StatUint64 hostMemUsedMB;
		struct GuestLibV3StatUint64 hostMemPhysMB;
		struct GuestLibV3StatUint64 hostMemPhysFreeMB;
		struct GuestLibV3StatUint64 hostMemKernOvhdMB;
		struct GuestLibV3StatUint64 hostMemMappedMB;
		struct GuestLibV3StatUint64 hostMemUnmappedMB;
		struct GuestLibV3StatUint32 memZippedMB;
		struct GuestLibV3StatUint32 memZipSavedMB;
		struct GuestLibV3StatUint32 memLLSwappedMB;
		struct GuestLibV3StatUint32 memSwapTargetMB;
		struct GuestLibV3StatUint32 memBalloonTargetMB;
		struct GuestLibV3StatUint32 memBalloonMaxMB;
		struct GuestLibV3ByteArray resourcePoolPathLong;
	} GuestLibV3Stat_u;
};
typedef struct GuestLibV3Stat GuestLibV3Stat;

/* the xdr functions */

#if defined(__STDC__) || defined(__cplusplus)
extern  bool_t xdr_GuestLibV3StatUint32 (XDR *, GuestLibV3StatUint32*);
extern  bool_t xdr_GuestLibV3StatUint64 (XDR *, GuestLibV3StatUint64*);
extern  bool_t xdr_GuestLibV3String (XDR *, GuestLibV3String*);
extern  bool_t xdr_GuestLibV3ByteArray (XDR *, GuestLibV3ByteArray*);
extern  bool_t xdr_GuestLibV3StatCount (XDR *, GuestLibV3StatCount*);
extern  bool_t xdr_GuestLibV3TypeIds (XDR *, GuestLibV3TypeIds*);
extern  bool_t xdr_GuestLibV3Stat (XDR *, GuestLibV3Stat*);

#else /* K&R C */
extern bool_t xdr_GuestLibV3StatUint32 ();
extern bool_t xdr_GuestLibV3StatUint64 ();
extern bool_t xdr_GuestLibV3String ();
extern bool_t xdr_GuestLibV3ByteArray ();
extern bool_t xdr_GuestLibV3StatCount ();
extern bool_t xdr_GuestLibV3TypeIds ();
extern bool_t xdr_GuestLibV3Stat ();

#endif /* K&R C */

#ifdef __cplusplus
}
#endif

#endif /* !_GUESTLIBV3_H_RPCGEN */
